ext/webidl/00_webidl.js

```diff
@@ -844,8 +846,21 @@
           opts,
         );
       }
-      const keys = ReflectOwnKeys(V);
       const result = {};
+      // Fast path for common case (not a Proxy)
+      if (!core.isProxy(V)) {
+        for (const key in V) {
+          if (!ObjectPrototypeHasOwnProperty(V, key)) {
+            continue;
+          }
+          const typedKey = keyConverter(key, opts);
```

Comment:
Would it be worth it to extract the three lines of code here into a helper function to avoid repetition?

Reply:
This is a very hot path, so extraneous function calls should be avoided.

Reply:
v8 would probably inline it, but lets not prematurely generalize